### PR TITLE
Rework the team invite modal functionality on members page [SCI-5099]

### PR DESCRIPTION
**Pull request by artoscinote**
Jira ticket: [SCI-5099](https://biosistemika.atlassian.net/browse/SCI-5099)

### What was done
Added multi-team invites to members page.

### CODE REVIEW

File: app/assets/javascripts/sitewide/dropdown_selector.js [hunk 1]
```diff
@@ -10,7 +10,7 @@
     data-placeholder // Search placeholder
     data-select-hint // A hint on top of a dropdown
     data-disable-on-load // Disable input after initialization
-    data-select-all-button // Text for select all button
+    data--button // Text for select all button
```
Comment (okriuchykhin) [inline app/assets/javascripts/sitewide/dropdown_selector.js:13]: what happened here?

Comment (artoscinote) [reply app/assets/javascripts/sitewide/dropdown_selector.js:13]: no idea, will revert.

File: app/controllers/users/invitations_controller.rb [hunk 2]
```diff
@@ -185,12 +198,18 @@ def check_captcha_for_invite
     def check_invite_users_permission
       @user = current_user
       @emails = params[:emails]&.map(&:downcase)
-      @team = Team.find_by_id(params['teamId'])
+
+      if params['teamId']
+        @teams = [Team.find(params['teamId'])]
+        return render_403 unless can_manage_team_users?(@teams.first)
```
Comment (okriuchykhin) [inline app/controllers/users/invitations_controller.rb:204]: Doesn't 'find' return only one record?

File: app/controllers/users/invitations_controller.rb [hunk 3]
```diff
@@ -185,12 +198,18 @@ def check_captcha_for_invite
     def check_invite_users_permission
       @user = current_user
       @emails = params[:emails]&.map(&:downcase)
-      @team = Team.find_by_id(params['teamId'])
+
+      if params['teamId']
+        @teams = [Team.find(params['teamId'])]
```
Comment (okriuchykhin) [inline app/controllers/users/invitations_controller.rb:203]: may be it's better to use 'where' to get ActiveRecord::Relation?

Comment (artoscinote) [reply app/controllers/users/invitations_controller.rb:203]: agreed

File: app/assets/javascripts/users/settings/teams/invite_users_modal.js [hunk 4]
```diff
@@ -46,7 +48,7 @@
       labelHTML: true,
       tagClass: 'users-dropdown-list',
       inputTagMode: true,
-      customDropdownIcon: () => { return '<i class="fas fa-search pull-right"></i>'; },
+      customDropdownIcon: () => { return '<i class="fas fa-search right-icon" style="display: block"></i>'; },
```
Comment (okriuchykhin) [inline app/assets/javascripts/users/settings/teams/invite_users_modal.js:51]: why did you put 'style' here?

Comment (artoscinote) [reply app/assets/javascripts/users/settings/teams/invite_users_modal.js:51]: will move to a css file

File: app/views/shared/_invite_users_modal.html.erb [hunk 5]
```diff
@@ -74,29 +74,40 @@ invite_to_team = type.in?(%w(invite_to_team invite_to_team_with_role))
...
-                                   .where('user_teams.role': UserTeam.roles[:admin])
-                                   .select { |team|  can_invite_team_users?(team) } %>
-            <% if teams.any? %>
-              <div class="team-selector">
-                <div class="heading">
-                  <input type="checkbox" data-role="team-selector-checkbox" />
-                  <span><%= t('invite_users.invite_to_team_heading') %></span>
+            <select class="team-ids-input" name="team_ids[]" style="display: none" multiple>
+            </select>
+            <div id="team-selector" class="team-selector">
```
Comment (okriuchykhin) [inline app/views/shared/_invite_users_modal.html.erb:84]: the same ID as class name?

File: app/controllers/users/invitations_controller.rb [hunk 6]
```diff
@@ -185,12 +198,18 @@ def check_captcha_for_invite
     def check_invite_users_permission
       @user = current_user
       @emails = params[:emails]&.map(&:downcase)
-      @team = Team.find_by_id(params['teamId'])
+
+      if params['teamId']
```
Comment (okriuchykhin) [inline app/controllers/users/invitations_controller.rb:203]: shouldn't it be 'team_ids'?

Comment (artoscinote) [reply app/controllers/users/invitations_controller.rb:203]: this is for the case of single team invite modal, from the teams section in admin

File: app/assets/javascripts/users/settings/teams/invite_users_modal.js [hunk 7]
```diff
@@ -26,6 +26,8 @@
     var teamSelectorDropdown = modal.find('[data-role=team-selector-dropdown]');
     var teamSelectorDropdown2 = $();
     var emailsInput = modal.find('.emails-input');
+    var teamsInput = modal.find('[name="team_ids[]"]');
```
Comment (okriuchykhin) [inline app/assets/javascripts/users/settings/teams/invite_users_modal.js:29]: why do you need another input with team ids?

File: app/views/shared/_invite_users_modal.html.erb (hunk 5 above)
Comment (artoscinote) [reply app/views/shared/_invite_users_modal.html.erb:84]: Is this an issue? Classes are usually used for CSS, IDs for JS targeting.

File: app/assets/javascripts/users/settings/teams/invite_users_modal.js [hunk 8]
```diff
@@ -181,6 +185,13 @@
       script.src = 'https://www.google.com/recaptcha/api.js?hl=en';
       $(script).insertAfter('#recaptcha-service');
       // Remove 'data-invited="true"' status
+
+      dropdownSelector.init(teamsInput, {
+        optionClass: 'checkbox-icon',
+        ajaxParams: function() {},
```
Comment (okriuchykhin) [inline app/assets/javascripts/users/settings/teams/invite_users_modal.js:191]: why do you need empty functions?

**Review by okriuchykhin — APPROVED** [review]

**Pull request merged**